orchestrator/api/api_v1/endpoints/subscriptions.py

```diff
@@ -103,41 +105,34 @@ def subscriptions_all() -> List[SubscriptionTable]:
 
 
 @router.get("/domain-model/{subscription_id}", response_model=SubscriptionDomainModelSchema)
-def subscription_details_by_id_with_domain_model(subscription_id: UUID) -> Dict[str, Any]:
-    customer_descriptions = SubscriptionCustomerDescriptionTable.query.filter(
-        SubscriptionCustomerDescriptionTable.subscription_id == subscription_id
-    ).all()
+def subscription_details_by_id_with_domain_model(
+    request: Request, subscription_id: UUID, response: Response
+) -> Optional[Dict[str, Any]]:
+    if_none_match = request.headers.get("If-None-Match")
+    if cache_response := from_redis(subscription_id):
+        domain_model, etag = cache_response
+        if etag == if_none_match:
+            response.status_code = HTTPStatus.NOT_MODIFIED
+            return None
+        response.headers["ETag"] = _generate_etag(domain_model)
+        return domain_model
+
     try:
-        subs_obj = SubscriptionModel.from_subscription(subscription_id)
+        subscription_model = SubscriptionModel.from_subscription(subscription_id)
+        extended_model = build_extendend_domain_model(subscription_model)
+        etag = _generate_etag(extended_model)
+
+        if etag == if_none_match:
```

Comment:
You could probably reshuffle this somewhat (use an else for the non-cached response) and have the etag setting outside the if/else to avoid some code duplication